manifests/record/mx.pp

```diff
@@ -1,15 +1,36 @@
 # == Define: dns::record::mx
 #
-# Wrapper for dns::record to set an XM record.
+# Wrapper for dns::record to set an MX record.
 #
 define dns::record::mx (
   $zone,
   $data,
-  $ttl = '',
-  $preference = '0',
-  $host = $name ) {
+  $ttl        = '',
+  $preference = 10,
+  $host       = '@' ) {
 
-  $alias = "${host},MX,${zone}"
+  $alias = "${host},${zone},MX,${preference},${data}"
+
+  unless is_string($zone) and is_domain_name($zone) and !is_ip_address($zone) and $zone !~ /\.?\d+$/ {
+    fail("Define[dns::record::mx]: MX zone ${zone} must be a valid domain name.")
+  }
+  unless is_string($data) and is_domain_name($data) and !is_ip_address($data) {
+    fail("Define[dns::record::mx]: MX data ${data} must be a valid hostname.")
+  }
+  unless $ttl =~ /^[0-9SsMmHhDdWw]+$/ or $ttl == '' {
```

Comment:
This TTL test is not specific to MX records. It should validate all BIND records, so would be better placed in the dns::record define.